src/Microsoft.Diagnostics.ExtensionCommands/Output/ColumnKind.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Diagnostics.ExtensionCommands.Output
+{
+    internal static class ColumnKind
+    {
+        private static Column? s_text;
+        private static Column? s_hexOffset;
+        private static Column? s_hexValue;
+        private static Column? s_dumpObj;
+        private static Column? s_integer;
+        private static Column? s_dumpHeapMT;
+        private static Column? s_listNearObj;
+        private static Column? s_dumpDomain;
+        private static Column? s_thread;
+        private static Column? s_integerWithoutComma;
+        private static Column? s_humanReadable;
+        private static Column? s_range;
+
+        private static int PointerLength => IntPtr.Size * 2;
```

Comment:
I'm not comfortable with this assumption that the target pointer size (IMemoryService.PointerSize) is the same as the size of the running process's pointer size.  Other than the DAC's requirement that it is loaded in the same-bitness as the target process, the rest of the infrastructure does make this assumption. I guess this is looking forward to the DCR managed "DAC" that wouldn't have the same-bitness requirement.  

Using the target pointer size here would be a pain so we can probably let this go for now with an NOTE/BUGBUG about this assumption. Maybe an assert in the Runtime.cs's code (where we have/get an instance of the memory service) that loads the DAC (IMemoryService.PointerSize == IntPtr.Size * 2).

Reply:
Let me think about it before merging this PR.  I'm not sure of a good way to get the pointer size without turning all Table related stuff into its own service.  That might be a good idea anyway...but it means that doing tricks like `using static Microsoft.Diagnostics.ExtensionCommands.Output.ColumnKind;` to reduce the wordiness of the code will be difficult.

I'll play around with it this week and see what I come up with.

Reply:
I've opted to resolve this with a NOTE/BUGBUG for now.  There's probably a way to rewrite a good chunk of this code to be a `TableService` which has `CreateTable` and since we are within a service we would be able to ServiceImport IMemoryService.

This all is doable, but I'm trying to preserve the code pattern of being able to write something like:

`Table foo = TableService.Create(Pointer, ByteCount, ...)`

But the `using static` trick only works with static properties.  Those static properties don't have a way to [ServiceImport] IMemoryService to get the pointer size.

I could just abandon this code pattern and require something like:

`Table foo = TableService.Create(TableService.Pointer, TableService.ByteCount, ...)`

Which works fine, but it gets really wordy, really quickly.

There's got to be a design that keeps things concise, but I haven't figured it out yet.

In any case, I'm running out of time on this particular project and I need to go work on test failures.  I'd like to get this one checked in, if you are ok with us having a TODO to rework this slightly, or incorporate Table deeper into the codebase at a later date.  Redesigning this all as a service will likely take up the rest of my week.

Reply:
I'm fine with checking this in the way it is. One thought: instead of static properties use an enum that looks up the proper formatting.  As I was writing this that I realized that it would be as wordy as about. Never mind.

Reply:
Also note that ClrMD takes a hard dependency on IntPtr.Size == IMemoryReader.PointerSize.

This isn't the _intended_ design, and if we ever have a runtime with real cross bitness I can update the codebase.  For now though, it wasn't worth worrying about since it's never been a real scenario (so far).

Reply:
> One thought: instead of static properties use an enum that looks up the proper formatting. As I was writing this that I realized that it would be as wordy as about

Yeah, that works with the `using static` syntax, but it being an enum means you can't do things like "Text.WithWidth(8)".

Reply:
FYI, the DCR SOS work for NativeAOT (from the recent DCR experiment) I'm going to finish up someday could do cross-bitness.

Reply:
Go ahead and check this in.